### PR TITLE
Tutorial Entry - Binary Search Tree

### PR DESCRIPTION
**Pull request by ChiragAgg5k**
## Change Summary

Wrote tutorial for Binary Search Trees. Closes issue #62 

### General 

- [x] This Pull Request is all my own work. (You'll be blacklisted if you are caught for plagiarism.)
- [x] I've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md)
- [x] I've applied LaTex for all variables, formulas and time / space complexity instead of using backticks
- [x] I've started the app locally and verified all the content and all links (if applicable) are accessible correctly
- [x] I've included Complexity Analysis (Time Complexity & Space Complexity).
- [x] I've written my explanation well and it is easy to understand for beginners

### Tutorial

- [x] I've read and followed the [Tutorial Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#tutorial-template)
- [x] I've explained my topic well with 2 - 3 LC problems and no external problems are used. 
- [x] I've provided the full working solutions to the problems used in this tutorial.
- [x] I've provided suggested problems at the end with the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [x] I've given credits / references if I use external resources. (For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).)

### Solutions

- [ ] I've read and followed the [Solution Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#solution-template)
- [ ] I've formatted my code well with [K&R Coding style](https://gist.github.com/jesseschalken/0f47a2b5a738ced9c845#why-kr)
- [ ] I've confirmed that comments are put above each line rather than writing on the same line.
- [ ] I've included a meaningful approach name for my solution. e.g. `## Approach 1: Two Pointers`.

Please do let me know if i missed something or got something wrong. Thanks!

### CODE REVIEW

**Review by wingkwong — CHANGES_REQUESTED** [review]
> A binary search tree (BST) is a [binary tree](http://localhost:3000/tutorials/graph-theory/binary-tree) where each node has a comparable key (and an associated value) and satisfies the restriction that the key in any node is larger than the keys in all nodes in that node's left subtree and smaller than the keys in all nodes in that node's right subtree.

Exactly same as https://algs4.cs.princeton.edu/32bst/

>  I've explained my topic well with 2 - 3 LC problems and no external problems are used.

you haven't used any LC problems to explain .

**Comment by ChiragAgg5k** [comment]
Okay working on it

**Comment by ChiragAgg5k** [comment]
@wingkwong i have added the changes

**Review by wingkwong — CHANGES_REQUESTED** [review]
your examples don't match with your tutorial.

**Comment by ChiragAgg5k** [comment]
Can you please explain what am i missing? Should i write a more detailed example? or pick a different question in the examples?

**Comment by wingkwong** [comment]
Your tutorials talk about insertion, deletion, searching but your examples don't relate to them directly.

**Comment by ChiragAgg5k** [comment]
Okay i will try fixing that

**Comment by ChiragAgg5k** [comment]
@wingkwong i changed the implementation of bst to match what leetcode uses on its site, and changed the examples as well. do let me know if any other probs

**Comment by ChiragAgg5k** [comment]
@wingkwong done

File: tutorials/graph-theory/binary-search-tree.md [hunk 1]
```diff
@@ -133,39 +177,59 @@ public:
...
+  public: TreeNode * searchBST(TreeNode * root, int key) {
+    // Key found
+    if (root == NULL || root -> val == key)
+      return root;
+
+    // Key is present in left subtree
+    if (key < root -> val)
+      return searchBST(root -> left, key);
+
+    // Else key is present in right subtree
```
Comment (wingkwong) [inline tutorials/graph-theory/binary-search-tree.md:228]: same as above.

**Review by wingkwong — CHANGES_REQUESTED** [review]
- `root -> left` -> `root->left` (apply to right and val)
- `temp -> left` -> `temp->left` (apply to right and val)
- `vector < int >` -> `vector<int>`
- `vector <int>` -> `vector<int>`
- for all `if` and `while`, please wrap with a `{}`.

**Comment by ChiragAgg5k** [comment]
@wingkwong done (2)

**Review by wingkwong — APPROVED** [review]

**Pull request merged**